File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    const losts = lost.filter(v=>!reserve.includes(v)).sort((a,b)=>a-b)
    const reserves =reserve.filter(v=>!lost.includes(v)).sort((a,b)=>a-b)
    
    for(let i = 0; i<losts.length; i++){
        for(let j = 0 ; j<reserves.length;j++){
            if(losts[i]-1===reserves[j] || losts[i]+1===reserves[j])
                {
                    losts.shift()
                    console.log(losts)
                }
        }
    }
    return n-losts.length;
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        
        
        List<Integer> lostList = new ArrayList<>();
        List<Integer> reserveList = new ArrayList<>();
        
        for (int l : lost) {
            boolean isInReserve = false;
            for (int r : reserve) {
                if (l == r) {
                    isInReserve = true;
                    break;
                }
            }
            if (!isInReserve) {
                lostList.add(l);
            }
        }

        for (int r : reserve) {
            boolean isInLost = false;
            for (int l : lost) {
                if (r == l) {
                    isInLost = true;
                    break;
                }
            }
            if (!isInLost) {
                reserveList.add(r);
            }
        }
        
        // 정렬
        Collections.sort(lostList);
        Collections.sort(reserveList);
        
        // 여분 체육복을 빌려줌
        for (int i = 0; i < lostList.size(); i++) {
            for (int j = 0; j < reserveList.size(); j++) {
                if (lostList.get(i) - 1 == reserveList.get(j) || lostList.get(i) + 1 == reserveList.get(j)) {
                    reserveList.remove(j);
                    lostList.remove(i);
                    i--; // 요소를 제거했으므로 인덱스를 조정
                    break;
                }
            }
        }
        
        return n - lostList.size();
        
    }
}